python/whylogs/core/metrics/condition_count_metric.py

```diff
@@ -0,0 +1,168 @@
+            raise ValueError("Condition cannot be named 'total'")
+
+        for cond_name in self.conditions.keys():
+            if cond_name not in self.matches:
+                self.matches[cond_name] = SumIntegralComponent(0)
+
+    def add_conditions(self, conditions: Dict[str, Condition]) -> None:
+        if "total" in conditions.keys():
+            raise ValueError("Condition cannot be named 'total'")
+        for cond_name, cond in conditions.items():
+            self.conditions[cond_name] = cond
+            self.matches[cond_name] = SumIntegralComponent(0)
+
+    def get_component_paths(self) -> List[str]:
+        paths: List[str] = [
+            "total",
+        ] + list(self.conditions.keys())
+        return paths
+
+    def columnar_update(self, data: PreprocessedColumn) -> OperationResult:
+        if data.len <= 0:
+            return OperationResult.ok(0)
+
+        count = 0
+        failed_conditions: Set[str] = set()
+        for x in list(chain.from_iterable(data.raw_iterator())):
+            count += 1
+            for cond_name, condition in self.conditions.items():
+                try:
+                    if condition.relation(x):
+                        self.matches[cond_name].set(self.matches[cond_name].value + 1)
+                    else:
+                        failed_conditions.add(cond_name)
+
+                except:  # noqa
+                    pass
+
+        self.total.set(self.total.value + count)
+        if condition.log_on_failure:
+            logger.warning(f"Condition(s) {', '.join(failed_conditions)} failed")
+        if condition.throw_on_failure:
+            raise ValueError(f"Condition(s) {', '.join(failed_conditions)} failed")
+        return OperationResult.ok(count)
+
+    @classmethod
+    def zero(cls, config: MetricConfig) -> "ConditionCountMetric":
+        if config is None or not isinstance(config, ConditionCountConfig):
+            raise ValueError("ConditionCountMetric.zero() requires ConditionCountConfig argument")
+
+        return ConditionCountMetric(
+            conditions=copy(config.conditions),
+            total=SumIntegralComponent(0),
+        )
+
+    def to_protobuf(self) -> MetricMessage:
+        msg = {"total": self.total.to_protobuf()}
+        for cond_name in self.conditions.keys():
+            msg[cond_name] = self.matches[cond_name].to_protobuf()
+
+        return MetricMessage(metric_components=msg)
+
+    def to_summary_dict(self, cfg: SummaryConfig) -> Dict[str, Any]:
+        summary = {"total": self.total.value}
+        for cond_name in self.conditions.keys():
+            summary[cond_name] = self.matches[cond_name].value
+
+        return summary
+
+    @classmethod
+    def from_protobuf(cls, msg: MetricMessage) -> "ConditionCountMetric":
+        cond_names: Set[str] = set(msg.metric_components.keys())
+        cond_names.remove("total")
+
+        conditions = {cond_name: re.compile("") for cond_name in cond_names}
```

Comment:
Like the other metrics, the configuration is lost on serialization but it's mergeable on deserialization. I will need to write a `merge()` for the metric. I can write it so that it clones the conditions from the receiving object--perhaps that suffices to handle distributed environments.

The `re.compile("")` was a left-over from when it initially only supported regex conditions.